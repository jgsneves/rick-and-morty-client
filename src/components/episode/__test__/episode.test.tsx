import { render } from '@testing-library/react';
import Episode from '..';

describe('Episode test suite', () => {
  it('should render only 11 characteres', () => {
    const { asFragment } = render(
      <Episode
        airDate="December 2, 2013"
        title="Pilot"
        characters={[
          'Rick Sanchez',
          'Morty Smith',
          'Bepisian',
          'Beth Smith',
          'Canklanker Thom',
          'Davin',
          'Frank Palicky',
          'Glenn',
          'Hookah Alien',
          'Jerry Smith',
          'Jessica',
          "Jessica's Friend",
          'Mr. Goldenfold',
          'Mrs. Sanchez',
          'Principal Vagina',
        ]}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render "+Mais" label', () => {
    const { getByText } = render(
      <Episode
        airDate="December 2, 2013"
        title="Pilot"
        characters={[
          'Rick Sanchez',
          'Morty Smith',
          'Bepisian',
          'Beth Smith',
          'Canklanker Thom',
          'Davin',
          'Frank Palicky',
          'Glenn',
          'Hookah Alien',
          'Jerry Smith',
          'Jessica',
          "Jessica's Friend",
          'Mr. Goldenfold',
          'Mrs. Sanchez',
          'Principal Vagina',
        ]}
      />,
    );
    const chars = getByText(
      /rick sanchez - morty smith - bepisian - beth smith - canklanker thom - davin - frank palicky - glenn - hookah alien - jerry smith - jessica; \+mais/i,
    );
    expect(chars).toBeInTheDocument();
  });
  it('should render 3 characteres', () => {
    const { asFragment } = render(
      <Episode
        airDate="December 2, 2013"
        title="Pilot"
        characters={['Rick Sanchez', 'Morty Smith', 'Bepisian']}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
