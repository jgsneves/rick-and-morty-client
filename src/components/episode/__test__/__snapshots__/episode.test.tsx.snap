// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode test suite should render 3 characteres 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ gzGsVq"
  >
    <h2
      class="sc-gsnTZi eTEMAC"
    >
      Pilot
    </h2>
    <p
      class="sc-dkzDqf jdgVjo"
    >
      Data de lançamento: 
    </p>
    <p
      class="sc-hKMtZM hyIBXv"
    >
      December 2, 2013
    </p>
    <p
      class="sc-dkzDqf jdgVjo"
    >
      Personagens: 
    </p>
    <p
      class="sc-hKMtZM hyIBXv"
    >
      Rick Sanchez - Morty Smith - Bepisian
    </p>
  </div>
</DocumentFragment>
`;

exports[`Episode test suite should render only 11 characteres 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ gzGsVq"
  >
    <h2
      class="sc-gsnTZi eTEMAC"
    >
      Pilot
    </h2>
    <p
      class="sc-dkzDqf jdgVjo"
    >
      Data de lançamento: 
    </p>
    <p
      class="sc-hKMtZM hyIBXv"
    >
      December 2, 2013
    </p>
    <p
      class="sc-dkzDqf jdgVjo"
    >
      Personagens: 
    </p>
    <p
      class="sc-hKMtZM hyIBXv"
    >
      Rick Sanchez - Morty Smith - Bepisian - Beth Smith - Canklanker Thom - Davin - Frank Palicky - Glenn - Hookah Alien - Jerry Smith - Jessica; +Mais
    </p>
  </div>
</DocumentFragment>
`;
